### PR TITLE
Generating documentation: clarify the explanation of `index.mld`

### PR DESCRIPTION
**Pull request by gasche**
I recently created a [Discussion topic](https://discuss.ocaml.org/t/how-to-get-pleasant-documentation-for-a-library-using-dune/7121) on how to write documentation with Dune, because I did not understand how to (easily) get a pleasant entry page for the project. The trick of providing an `index.mld` file was explained in this topic.

This PR modifies the Dune documentation to clarify the use of `index.mld` file. In fact, this feature/trick was already mentioned in the documentation, but only in the reference documentation of the `(documentation)` stanza, not in the "Generating documentation" page. I had read the "Generating documentation" page carefully and yet I did not know about using `index.mld`.

With this PR:
- The explanation of `index.mld` is moved from the stanza documentation to the "Generating documentation" page, with some clarifications and additional emphasis.
- The "Generating documentation" page points to the stanza documentation
- The stanza documentation explicitly points to the "Generating documentation" page

### CODE REVIEW

File: doc/documentation.rst [hunk 1]
```diff
@@ -79,6 +80,24 @@ directory to the inferred package:
...
+Package entry page
+------------------
+
+The ``index.mld`` file (specified as ``index`` in ``mld_files``) is
+treated specially by dune. This will be the file used to generate the
+entry page for the package, linked from the main package listing.
+
+To generate pleasant documentation, we recommend writing an
+``index.mld`` file with at least short description of your package,
+and possibly a couple examples.
```
Comment (nojb) [inline doc/documentation.rst:95]:
```suggestion
and possibly a couple of examples.
```

**Review by nojb — APPROVED** [review]
Thanks, documentation PRs are always great :) Could you please sign your commit (`git commit -s`) ?

File: doc/documentation.rst (hunk 1 above)
Comment (gasche) [reply doc/documentation.rst]: I used `possibly some examples.`

**Comment by gasche** [comment]
Thanks! I signed the commits off, and took the comment into account.

**Comment by nojb** [comment]
Thanks! Merged.

**Pull request merged**